graspy/align/seedless_procrustes.py

```diff
@@ -0,0 +1,373 @@
+            reg=self.optimal_transport_lambda,
+            numItermax=self.optimal_transport_eps,
+            stopThr=self.optimal_transport_eps,
+        )
+        return P
+
+    def _procrustes(self, X, Y, P_i):
+        # "M step" of the SeedlessProcurstes.
+        aligner = OrthogonalProcrustes()
+        Q = aligner.fit(X, P_i @ Y).Q_X
+        return Q
+
+    def _iterative_ot(self, X, Y, Q):
+        for i in range(self.iterative_num_reps):
+            P_i = self._optimal_transport(X, Y, Q)
+            Q = self._procrustes(X, Y, P_i)
+            c = np.linalg.norm(X @ Q - P_i @ Y, ord="fro")
+            if c < self.iterative_eps:
+                break
+        return P_i, Q
+
+    def fit(self, X, Y):
+        """
+        Uses the two datasets to learn matrices Q_X and Q_Y.
+        In seedless procrustes Q_X is a final solution of the to the iterative
+        optimal transport / procrustes algorithm and Q_Y is the identity
+        matrix.
+
+        Parameters
+        ----------
+        X: np.ndarray, shape (n, d)
+            First dataset of vectors. These vectors need to have same number of
+            dimensions as ones in Y, but the number of vectors can differ.
+
+        Y: np.ndarray, shape (m, d)
+            Second dataset of vectors. These vectors need to have same number
+            of dimensions as ones in X, but the number of vectors can differ.
+
+        Returns
+        -------
+        self: returns an instance of self
+        """
+
+        # check for numpy-ness, 2d-ness and finite-ness
+        if not isinstance(X, np.ndarray):
+            msg = f"first dataset is a {type(X)}, not an np.ndarray! "
+            raise TypeError(msg)
+        if not isinstance(Y, np.ndarray):
+            msg = f"first dataset is a {type(Y)}, not an np.ndarray! "
+            raise TypeError(msg)
+        X = check_array(X, accept_sparse=True, copy=True)
+        Y = check_array(Y, accept_sparse=True, copy=True)
+
+        # check for equal components and number of entries
+        if X.shape[1] != Y.shape[1]:
+            msg = "two datasets have different number of components!"
+            raise ValueError(msg)
+        _, d = X.shape
+
+        if self.initialization == "2d":
+            n, d = X.shape
+            m, _ = Y.shape
+            P_matrices = np.zeros((2 ** d, n, m))
+            Q_matrices = np.zeros((2 ** d, d, d))
+            objectives = np.zeros(2 ** d)
+            # try 2^d different initializations
+            for i in range(2 ** d):
+                initial_Q = self._orthogonal_matrix_from_int(i, d)
+                P_matrices[i], Q_matrices[i] = P, Q = self._iterative_ot(
+                    X, Y, initial_Q
+                )
+                objectives[i] = np.linalg.norm(X @ Q - P @ Y, ord="fro")
+            # pick the best one, using the objective function value
+            best = np.argmin(objectives)
+            self.initial_Q = self._orthogonal_matrix_from_int(best, d)
```

Comment:
@bdpedigo a reminder to think about this, overloading the input parameter seems odd but this could be useful information to have

Reply:
note that the only times it is "overloaded" is when it was irrelevant to begin with: i.e. when init was "2d" or "sign-flips".

Reply:
FYI, overwriting initial parameters is not allowed if sklearn compliance is required..

Reply:
okay; this is a fair point. the question is what do we want to do? i can have initial_Q and initial_Q_, but this seems confusing. i can just not overload it at all, though i do agree with Ben, it seems like a very useful information to have. how much do we care about being sklearn compliant?

Reply:
idk if we are not including this to `tests/test_sklearn.py`, then i would say this is ok

Reply:
okay. i'm still happy to change this to something better if you and/or Ben think of such.

Reply:
now throws errors if initial_Q or initial_P was passed but intiialization was not custom, so it can only overwrite Nones.

Reply:
not 100% kosher, but i'd say like 90%. closing this.